paprika/build.py

```diff
@@ -1,185 +1,141 @@
-import logging as log
-import numpy as np
 import os as os
 import re as re
 import subprocess as sp
+
+import logging as log
+import numpy as np
 import parmed as pmd
 from parmed.structure import Structure as ParmedStructureClass
 from paprika import utils
 
+N_A = 6.0221409 * 10**23
+ANGSTROM_CUBED_TO_LITERS = 1 * 10**-27
 
-from .utils import check_for_leap_log
-
-def add_dummy(structure, atom_name='DUM', residue_name='DUM', mass=208.00, atomic_number=82, x=0.000, y=0.000, z=0.000):
-    """ Add a dummy atom at the specified coordinates to the end of the structure """
-
-    # Assume the input_structure is a filename or ...
-    if type(structure) is str:
-        structure = utils.return_parmed_structure(structure)
-    # ... a parmed.structure
-    elif type(structure) is ParmedStructureClass:
-        pass
-    else:
-        raise Exception('add_dummy does not support the type associated with structure:'+type(structure))
-
-    # Create an atom object
-    dum = pmd.topologyobjects.Atom()
-    dum.name = atom_name
-    dum.mass = mass
-    dum.atomic_number = atomic_number
-    dum.xx = x
-    dum.xy = y
-    dum.xz = z
-    
-    # Assume that the atom numbering we read in is correct!!
-    dum.number = structure.atoms[-1].number + 1
-    residue_num = structure.residues[-1].number + 1
-    
-    # Add to structure
-    structure.add_atom(dum, residue_name, residue_num)
-
-    # tleap will probably want TER cards in any PDBs we make, so enforce
-    # that for both the dummy residue and the residue before it
-    structure.residues[-2].ter = True
-    structure.residues[-1].ter = True
-
-    return structure
-
-def write_dummy_frcmod(path='.', filename='dummy.frcmod', atomtype='Du', mass='208.00'):
-    """ Write a frcmod file for a dummy atom """
-
-    with open(path+'/'+filename, 'w') as f:
-        f.write("""\
-Parameters for dummy atom with type {0}
-MASS
-{0}     {1}
-
-BOND
-
-ANGLE
-
-DIHE
-
-IMPROPER
-
-NONBON
-  {0}       0.000     0.0000000
-""".format(atomtype, mass))
-
-
-def write_dummy_mol2(path='.', filename='dummy.mol2', atom_name='DUM', atomtype='Du', residue_name='DUM'):
-    """ Write a mol2 file for a dummy atom """
-
-    with open(path+'/'+filename, 'w') as f:
-        f.write("""\
-@<TRIPOS>MOLECULE
-{0}
-    1     0     1     0     1
-SMALL
-USER_CHARGES
-
-@<TRIPOS>ATOM
-  1 {1:4s}    0.000000    0.000000    0.000000 {2}    1 {0}     0.0000 ****
-@<TRIPOS>BOND
-@<TRIPOS>SUBSTRUCTURE
-      1  {0}              1 ****               0 ****  ****    0 ROOT
-""".format(residue_name[0:3], atom_name, atomtype[0:2]))
 
+def default_tleap_options():
+    """
+    Return a dictionary of default `tleap` options.
+    Returns
+    -------
 
-def read_tleaplines(tleapfile, pdbfile=None, skip_solvate=True):
     """
-    Read a functional tleap file and return a list containing each line of instruction
+    options = {}
+    options['unit'] = 'model'
+    options['pbc_type'] = 1
```

Comment:
Let's change pbc_type from using integers to using keywords:  cubic, z-rectangular, truncated_octahedron

... or something in that spirit.  numbers are dumb and hard to remember, not sure why I wrote it that way.

Reply:
Sure, will do.

Reply:
Fixed in https://github.com/slochower/pAPRika/pull/43/commits/c6da408dfa655c3149d9e4c7fb832050f0553b32